### ISSUE TITLE
C++/CLI .NET Core 3.1 Injection to Native Process

### ISSUE THREAD

**Issue by johmarjac** [dotnet/runtime#33590]
Hello,

I have a question. On .NET Framework C++/CLI DLL's I was able to have a DllMain inside the C++/CLI library which was getting called upon attaching the Dll to a native process (aka injecting it to a native process).

With .NET Core C++/CLI Dll this does not seem to be working. DllMain is not getting called. Can anyone explain me why and if it possible to call it nevertheless upon injection or do I have to make a more stony way by exporting a function in the C++/CLI project, and iterate over the exported functions of the library upon injection to call it manually?

Edit: Calling LoadLibrary on a .NET Core C++/CLI seems to fail too with GetLastError() = 126
Edit2: Calling LoadLibraryEx with Flag DONT_RESOLVE_DEPENDENCIES worked. Exporting functions worked as welll and I was able to GetProcAddress my custom entry point now. 

However DllMain is myself still a puzzle.

**Comment by vitek-karas** [comment]
@jkoritzinsky

**Comment by jkoritzinsky** [comment]
Do you have the `ijwhost.dll` deployed along-side your C++/CLI assembly? The ijwhost.dll has the native implementation of the DLL entry-point to bootstrap the .NET Core runtime.

Can you try injecting the `ijwhost.dll` into the application before injecting your C++/CLI assembly?

**Comment by johmarjac** [comment]
Hi, thanks for the answer.
I did the following changes to the `CLR Core Class Library (.NET Core)` Template of VS2019:

Switched from x86 to x64 as my target process is a x64 process.
Then in the pch.h
![image](https://user-images.githubusercontent.com/30114127/76784602-39388100-67b4-11ea-9812-b1e3a3caa987.png)
I just added the windows header.

And in my "Main.cpp" file the template created it looks like that:
![image](https://user-images.githubusercontent.com/30114127/76784677-58371300-67b4-11ea-9dac-7321b409f033.png)

By that, I am expecting, upon injection, a console window appear, which gets free'd instantly after but will remain visible, closeable without closing the host process (that's at least how it works normally).

![image](https://user-images.githubusercontent.com/30114127/76784891-c4b21200-67b4-11ea-9ac3-ae8ba0ba5a7f.png)

I now put the compiled library in the same folder as the target executable just like the ijwhost.dll.
If I now inject the MemSharpNETHost.dll into the process nothing happens. x64Dbg does not show MemSharpNETHost.dll as dependency, neither does ijwhost.dll.
If I however do it like you suggested, first inject ijwhost.dll, then my MemSharpNETHost.dll, then ijwhost.dll showed up in the dependencies, as well as MemSharpNETHost.dll. 

Console however does never show up.

To make sure I attached Visual Studio to the remote process and put a breakpoint on DllMain:
![image](https://user-images.githubusercontent.com/30114127/76785689-2921a100-67b6-11ea-890e-b68ab1da66e0.png)

So I expected upon injection to hit the breakpoint there before creating the thread, but nope. It didnt hit the breakpoint, and the half-red-breakpoint circle also didnt change to full-red upon injection which it does with a normal native library injection when VS is attached.
